### PR TITLE
Implemented new reviewer chooser and delete dead code

### PR DESCRIPTION
**Pull request by kaedroho**

### CODE REVIEW

File: wagtail_review/admin_api/serializers.py [hunk 1]
```diff
@@ -1,16 +1,36 @@
+from django.contrib.auth import get_user_model
 from rest_framework import serializers
 
 from .. import models
 from ..api.serializers import CommentSerializer
 
 
+class UserSerializer(serializers.ModelSerializer):
+    id = serializers.ReadOnlyField(source='pk')
+    full_name = serializers.ReadOnlyField(source='get_full_name')
```
Comment (emilytoppm) [inline wagtail_review/admin_api/serializers.py]: Just noting that this is optional for a custom user model, as per https://docs.djangoproject.com/en/3.0/topics/auth/customizing/. Is this relied upon?

File: wagtail_review/static_src/admin/src/api/reviewer.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                credentials: 'same-origin'
+            }
+        );
+
+        return response.json();
+    }
+
+    async _newReviwer(body: string): Promise<NewReviewerResponse> {
+        let response = await fetch(`/admin/wagtail_review/api/reviewers/`, {
+            method: 'POST',
```
Comment (emilytoppm) [inline wagtail_review/static_src/admin/src/api/reviewer.ts:66]: The `AdminAPIViewMixin` relies on session authentication, which according to https://www.django-rest-framework.org/api-guide/authentication/#sessionauthentication will require a valid CSRF token for any POST requests as here. Is this attached somehow? Or does `FooAuth` circumvent this, in which case there's maybe a case for this being somewhat unsafe?

**Review by emilytoppm — CHANGES_REQUESTED** [review]
Generally looks great, but there's a few issues remaining (partly my fault, as we handed over mid-development!). 
Should be fixed: 
- `allow_responses` in the template tag is inconsistent with the `Respond` view: potentially allowing a reviewer to pick an action they don't actually have permission to perform in the API
- Tests need to be updated and run
To look at:
- UserSerializer
- CSRF issues

File: wagtail_review/static_src/admin/src/api/reviewer.ts (hunk 2 above)
Comment (kaedroho) [reply wagtail_review/static_src/admin/src/api/reviewer.ts:66]: The fact it's called `FooAuth` suggests to me this was only a temporary hack that I've forgotten about. Thanks for spotting!

Comment (kaedroho) [reply wagtail_review/static_src/admin/src/api/reviewer.ts:66]: Fixed in latest commit

File: wagtail_review/admin_api/serializers.py (hunk 1 above)
Comment (kaedroho) [reply wagtail_review/admin_api/serializers.py]: Fixed in latest commit

**Review by gasman — APPROVED** [review]
Latest round of fixes all look good to me!

File: wagtail_review/models.py [hunk 3]
```diff
@@ -345,23 +347,51 @@ class ReviewResponse(models.Model):
...
+            external_user = self.reviewer and self.reviewer.external
+
+            if external_user:
+                email_address = escape(external_user.email)
+                return mark_safe(f"""
+                    Reviewed by external user: <b>{email_address}</b><br/>
+                    <blockquote>{comment}</blockquote>
+                """)
+            else:
+                return mark_safe(f"<blockquote>{comment}</blockquote>")
```
Comment (gasman) [inline wagtail_review/models.py:391]: Is there a reason why this isn't using Django's `format_html`? It seems that would eliminate a lot of the explicit escaping and mark_safe (and avoid the python-3.5 incompatibility of format strings).

Comment (kaedroho) [reply wagtail_review/models.py:391]: Just didn't think of doing it that way, will change now. Thanks!

**Pull request closed without merging**